libraries/SD/src/utility/SdFile.cpp

```diff
@@ -905,7 +906,7 @@ uint8_t SdFile::rmRfStar(void) {
       if (!f.remove()) return false;
     }
     // position to next entry if required
-    if (curPosition_ != (32*(index + 1))) {
+    if (curPosition_ != (uint32_t)(32*(index + 1))) {

```

Comment:
index is uint16_t, and the value is being expanded.
I _think_ this is ok, after reading up on usual arithmetic conversions, but I'm not positive. Please double check that the cast is ok on the final arithmetic result, and that it shouldn't be done directly on index _before_ the arithmetic operations.

Reply:
Didn't mean to expand it, thought curPosition_ was 32-bits.  Will correct to uint16_t.

Reply:
Misread your comment, will move the *cast* inside.  No 16 bits.